### ISSUE TITLE
Hi, I have some questions about the "autoanchor.py"

### ISSUE THREAD

**Issue by xiaomao19970819** [ultralytics/yolov5#5614]
### Search before asking

- [ ] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

in yolov5/utils/autoanchor.py, i found
scale = np.random.uniform(0.9, 1.1, size=(shapes.shape[0], 1)) # augment scale
wh = torch.tensor(np.concatenate([l[:, 3:5] * s for s, l in zip(shapes * scale, dataset.labels)])).float() # wh
in the check_anchors() function.
Here, the scale factor (0.1) is different from the scale of training data augmentation (0.5). Is this reasonable?
Will this result in some bounding boxes being shrunk and then discarded because of the threshold（0.25，4）?

Also, in kmean_anchors() function, you have not used scaling factor calculations again, why is that?

### Additional

I hope I can get your answer. Thank you

**Comment by glenn-jocher** [comment]
@xiaomao19970819 hi, thank you for your feature suggestion on how to improve YOLOv5 🚀!

The fastest and easiest way to incorporate your ideas into the official codebase is to submit a Pull Request (PR) implementing your idea, and if applicable providing before and after profiling/inference/training results to help us understand the improvement your feature provides. This allows us to directly see the changes in the code and to understand how they affect workflows and performance.

Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.

**Comment by xiaomao19970819** [comment]
I didn't mean to suggest an improvement, I don't know if my post is correct, I wonder why you set the scaling factor in autoanchor

> @xiaomao19970819 hi, thank you for your feature suggestion on how to improve YOLOv5 🚀!
> 
> The fastest and easiest way to incorporate your ideas into the official codebase is to submit a Pull Request (PR) implementing your idea, and if applicable providing before and after profiling/inference/training results to help us understand the improvement your feature provides. This allows us to directly see the changes in the code and to understand how they affect workflows and performance.
> 
> Please see our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/) to get started.

I didn't mean to suggest an improvement, I don't know if my idea is correct, I just wonder why you set the scaling factor in autoanchor.py
Why is scale not used when generating  matching anchors for the dataset.

**Comment by glenn-jocher** [comment]
@xiaomao19970819 oh, scaling factor in auto anchor is there to avoid underdetermined kmeans problems, i.e. if a dataset has all objects same size kmeans can not initialize properly if the unique points are less than the requested points.

**Comment by xiaomao19970819** [comment]
I noticed that sacle parameter 0.5 was used in data augmentation. Does this result in the target bounding box being scaled not matching the anchor points generated before?

**Comment by xiaomao19970819** [comment]
> @xiaomao19970819 oh, scaling factor in auto anchor is there to avoid underdetermined kmeans problems, i.e. if a dataset has all objects same size kmeans can not initialize properly if the unique points are less than the requested points.

I noticed that sacle parameter 0.5 was used in data augmentation. Does this result in the target bounding box being scaled not matching the anchor points generated before?

**Comment by glenn-jocher** [comment]
@xiaomao19970819 no this does not affect autoanchor, at least not at the current anchor_t of 4.0:
https://github.com/ultralytics/yolov5/blob/69032519bc575ef6c2033ab0f7d9bc1f9651b251/data/hyps/hyp.scratch-low.yaml#L19
